### ISSUE TITLE
docker-compose does not see EOF on STDIN when used on TLS protected docker engine

### ISSUE THREAD

**Issue by zibok** [docker/compose#5535]
## How to reproduce

* docker 12.09.0-ce with TLS enabled
* docker-compose 1.8.1+ (1.8.0 is working correctly)
* docker-compose.yml as follow:
```yaml
version: '2'

services:
  test:
    image: debian:jessie
```
* Run the command:
```
echo toto | docker-compose run test tee toto.txt
```

## Expected behaviour

The command ends properly and the container's toto.txt contains "toto"

## Actual behaviour

The command hangs, though the container's toto.txt contains "toto"

## Additional information

```
$ docker info
Containers: 3
 Running: 0
 Paused: 0
 Stopped: 3
Images: 1
Server Version: 17.09.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.0-4-amd64
Operating System: Debian GNU/Linux 9 (stretch)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 492.4MiB
Name: stretch
ID: HJSC:OKNJ:FNOI:2G52:OYNL:KULV:C3P5:W5YC:Y4NX:LH2C:QZVB:SQCN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Comment by shin-** [comment]
That version of Compose is outdated. Please upgrade to 1.18.

**Comment by zibok** [comment]
Sorry, I haven't made myself clear enough. I first encountered the issue with 1.18.0, and since it worked with the 1.6 my script was using, I tried to figure out when it broke.
I tried 1.17.0, 1.13.0, 1.11.0, 1.10.0, 1.9.0 and 1.8.1. Each time it was hanging. But 1.8.0 was working great.
That's why I put a **+** after 1.8.1, it was to help in the investigations.

**Comment by shin-** [comment]
Ah, sorry, I missed that. Did you install those different versions using `pip`, or are you using the binary releases? And if using `pip`, were you updating all dependencies alongside Compose, or just the `docker-compose` package?

Thanks!

**Comment by zibok** [comment]
I use the binaries (downloaded via curl as stated in the install documentation)

**Comment by shin-** [comment]
Alright, thanks for the info. I'll see what I can do.

**Comment by shin-** [comment]
Should be fixed by #5580

**Comment by zibok** [comment]
Seems to work on master build. Thanks!
